lib/usecases/login_use_case.dart

```diff
@@ -0,0 +1,37 @@
+import 'package:injectable/injectable.dart';
+
+import '../api/exception/network_exceptions.dart';
+import '../api/repository/authentication_repository.dart';
+import '../local/local_storage.dart';
+import '../model/auth_token.dart';
+import 'base/base_use_case.dart';
+
+class LoginInput {
+  final String email;
+  final String password;
+
+  LoginInput({required this.email, required this.password});
+}
+
+@lazySingleton
+class LoginUseCase extends UseCase<void, LoginInput> {
+  final AuthenticationRepository _repository;
+  final LocalStorage _localStorage;
+
+  const LoginUseCase(this._repository, this._localStorage);
+
+  @override
+  Future<Result<void>> call(LoginInput params) {
+    return _repository
+        .logIn(email: params.email, password: params.password)
+        .then((token) => saveTokens(token))
+        .onError<NetworkExceptions>(
+            (err, stackTrace) => Failed(UseCaseException(err)));
+  }
+
+  Result<dynamic> saveTokens(AuthToken authToken) {
```

Comment:
```suggestion
  Result<void> saveTokens(AuthToken authToken) {
```

Reply:
@Wadeewee resolved in https://github.com/nimblehq/ic-flutter-avishek/pull/34/commits/9a8bd23566f92cf478aba25df9d9d71f41d70585